### PR TITLE
cloud_runner: prepare Dockerfile according to ccache

### PR DESCRIPTION
**Pull request by DavidKorczynski**

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
Need a sister PR on OSS-Fuzz that uses the ARG

**Comment by DavidKorczynski** [comment]
Needs https://github.com/google/oss-fuzz/pull/12693 to land to make this testable

File: experiment/oss_fuzz_checkout.py [hunk 1]
```diff
@@ -296,6 +296,47 @@ def is_image_cached(project_name: str, sanitizer: str) -> bool:
     return False
 
 
+def rewrite_project_to_cached_project_cloud_ccache(generated_project) -> None:
+  """Rewrites Dockerfile to work wiht ccache logic for cloud builds"""
```
Comment (oliverchang) [inline experiment/oss_fuzz_checkout.py:300]: nit: spelling of wiht.

File: experiment/oss_fuzz_checkout.py [hunk 2]
```diff
@@ -296,6 +296,47 @@ def is_image_cached(project_name: str, sanitizer: str) -> bool:
     return False
 
 
+def rewrite_project_to_cached_project_cloud_ccache(generated_project) -> None:
```
Comment (oliverchang) [inline experiment/oss_fuzz_checkout.py:299]: nit: is this ccache specific? Or just chronos? 

The exact same steps will work for the autogenerated replay build scripts right?

**Review by oliverchang — APPROVED** [review]

File: experiment/oss_fuzz_checkout.py (hunk 1 above)
Comment (DavidKorczynski) [reply experiment/oss_fuzz_checkout.py]: fixed

File: experiment/oss_fuzz_checkout.py (hunk 2 above)
Comment (DavidKorczynski) [reply experiment/oss_fuzz_checkout.py]: yes, changed the name

**Review by oliverchang — APPROVED** [review]

**Pull request merged**